lib/lualinux.c

```diff
@@ -105,6 +106,22 @@ static int lualinux_lookup(lua_State *L)
 	return 1;
 }
 
+static int lualinux_ifindex(lua_State *L)
+{
+	int ifindex;
+	const char *ifname = luaL_checklstring(L, 1, NULL);
+	struct net_device *dev = dev_get_by_name(&init_net, ifname);
+	if (!dev) {
+		lua_pushinteger(L, -ENODEV);
+		lua_error(L);
+	}
+	ifindex = dev->ifindex;
+	dev_put(dev);
+
+	lua_pushinteger(L, ifindex);
+	return 1;
+}
```

Comment:
```suggestion
static int lualinux_ifindex(lua_State *L)
{
	const char *ifname = luaL_checkstring(L, 1);
	struct net_device *dev = dev_get_by_name(&init_net, ifname);

	luaL_argcheck(L, dev != NULL, 1, "device not found");
	lua_pushinteger(L, dev->ifindex);
	dev_put(dev);
	return 1;
}
```

as we are not in the realm of luasocket anymore, I would raise a proper error message instead of throwing the `errno`.. also, I would use argcheck to both simplify the code and indicate that it's an invalid argument.. also, though it's not written code style, I would mind that we are using a `\n` after the declarations..